### PR TITLE
fix: Added scroll to top button

### PR DESCRIPTION
**Pull request by sumitkr2000**
## Related Issue

#1696 

## Description

Hey @priyankarpal I have added scroll to top button on the website. Please make this a level 2 pr as a new feature is being added. Please review my pr and merge it. Thank you!

## Screenshots

image

### CODE REVIEW

File: app/layout.tsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
+"use client"
```
Comment (priyankarpal) [inline app/layout.tsx:1]: remove this line
```suggestion
```

File: components/ScrollToTop.tsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React, { useEffect, useState } from 'react';
```
Comment (priyankarpal) [inline components/ScrollToTop.tsx:1]: add `"use client"` here 

```suggestion
"use client"
import React, { useEffect, useState } from 'react';
```

File: components/ScrollToTop.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import { FaArrowUp } from 'react-icons/fa';
+import { Link } from 'react-scroll';
```
Comment (priyankarpal) [inline components/ScrollToTop.tsx:3]: we don't need to use `Link` property
```suggestion
```

File: components/ScrollToTop.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+function BackToTopButton() {
+  const [btnVisiblity, setBtnVisiblity] = useState(false);
+  const handleScroll = () => {
+    window.scrollTo({ top: 0, behavior: 'smooth' });
+  };
+  useEffect(() => {
+    const toggleVisiblity = () => {
+      window.pageYOffset > 250 ? setBtnVisiblity(true) : setBtnVisiblity(false);
```
Comment (priyankarpal) [inline components/ScrollToTop.tsx:12]: `pageYOffset` is deprecated so use `scrollY` 

```suggestion
      window.scrollY > 250 ? setBtnVisiblity(true) : setBtnVisiblity(false);
```

File: components/ScrollToTop.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    <Link onClick={handleScroll} to="/" className="top">
+      <div
+        className={` ${!btnVisiblity ? 'translate-y-32 md:translate-y-20' : 'translate-y-0'
+          } fixed bottom-20 md:bottom-10 right-8 z-50 w-10 h-10 rounded-lg bg-primary text-white cursor-pointer flex justify-center items-center transform transition-all delay-100 duration-500 `}
+      >
+        <div className="button">
+          <FaArrowUp />
+        </div>
+      </div>
+    </Link>
```
Comment (priyankarpal) [inline components/ScrollToTop.tsx:30]: here also, rename to `button`
```suggestion
    </button>
```

File: package.json [hunk 6]
```diff
@@ -15,6 +15,7 @@
     "@types/node": "20.4.2",
     "@types/react": "18.2.15",
     "@types/react-dom": "18.2.7",
+    "@types/react-scroll": "^1.8.7",
```
Comment (priyankarpal) [inline package.json:18]: we don't need an extra dependencies for `scroll to top` just remove

File: package.json [hunk 7]
```diff
@@ -25,6 +26,7 @@
     "react-icons": "^4.10.1",
     "react-infinite-scroll-component": "^6.1.0",
     "react-router-dom": "^6.14.1",
+    "react-scroll": "^1.8.9",
```
Comment (priyankarpal) [inline package.json:29]: same here we don't need an extra dependencies for `scroll to top` just remove

**Review by priyankarpal — CHANGES_REQUESTED** [review]
after removing dependencies run `pnpm i` & push @sumitkr2000

**Comment by sumitkr2000** [comment]
@priyankarpal done!

**Review by priyankarpal — APPROVED** [review]
Looks Good 🎉

**Comment by sumitkr2000** [comment]
> Looks Good 🎉

thanks!

**Pull request merged**